algorithms/src/polycommit/kzg10/data_structures.rs

```diff
@@ -26,128 +26,81 @@ use snarkvm_utilities::{
     error,
     io::{Read, Write},
     serialize::{CanonicalDeserialize, CanonicalSerialize},
-    Compress,
     FromBytes,
-    SerializationError,
     ToBytes,
     ToMinimalBits,
-    Validate,
 };
 
 use anyhow::Result;
 use core::ops::{Add, AddAssign};
-use parking_lot::RwLock;
 use rand_core::RngCore;
-use std::{collections::BTreeMap, io, sync::Arc};
+use std::{collections::BTreeMap, io, ops::Range};
 
 /// `UniversalParams` are the universal parameters for the KZG10 scheme.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize)]
```

Comment:
What size of `PowersOfG` does Canonical* write to bytes?

Reply:
Whatever is in memory (it writes out the size of the existing powers)